<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="TC-tea">

<title>BIO312 Final Exam List</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="index_files/libs/clipboard/clipboard.min.js"></script>
<script src="index_files/libs/quarto-html/quarto.js"></script>
<script src="index_files/libs/quarto-html/popper.min.js"></script>
<script src="index_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="index_files/libs/quarto-html/anchor.min.js"></script>
<link href="index_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="index_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="index_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="index_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script async="" src="https://hypothes.is/embed.js"></script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#tool" id="toc-tool" class="nav-link active" data-scroll-target="#tool"><strong>Tool</strong></a></li>
  <li><a href="#courseware" id="toc-courseware" class="nav-link" data-scroll-target="#courseware"><strong>Courseware</strong></a>
  <ul>
  <li><a href="#lecture" id="toc-lecture" class="nav-link" data-scroll-target="#lecture">Lecture</a></li>
  <li><a href="#exercise" id="toc-exercise" class="nav-link" data-scroll-target="#exercise">Exercise</a></li>
  </ul></li>
  <li><a href="#code" id="toc-code" class="nav-link" data-scroll-target="#code"><strong>Code</strong></a>
  <ul>
  <li><a href="#de-novo-for-mz-plot" id="toc-de-novo-for-mz-plot" class="nav-link" data-scroll-target="#de-novo-for-mz-plot">De Novo for M/Z plot</a></li>
  <li><a href="#molecular-weight-of-mass-spectrum" id="toc-molecular-weight-of-mass-spectrum" class="nav-link" data-scroll-target="#molecular-weight-of-mass-spectrum">Molecular weight of mass spectrum</a></li>
  </ul></li>
  <li><a href="#past-exam-paper" id="toc-past-exam-paper" class="nav-link" data-scroll-target="#past-exam-paper"><strong>Past exam paper</strong></a>
  <ul>
  <li><a href="#bio312-final-exam-2022-2023" id="toc-bio312-final-exam-2022-2023" class="nav-link" data-scroll-target="#bio312-final-exam-2022-2023">BIO312 final exam (2022-2023)</a>
  <ul class="collapse">
  <li><a href="#question-1" id="toc-question-1" class="nav-link" data-scroll-target="#question-1">Question 1</a></li>
  <li><a href="#question-2" id="toc-question-2" class="nav-link" data-scroll-target="#question-2">Question 2</a></li>
  <li><a href="#question-3" id="toc-question-3" class="nav-link" data-scroll-target="#question-3">Question 3</a></li>
  <li><a href="#question-4" id="toc-question-4" class="nav-link" data-scroll-target="#question-4">Question 4</a></li>
  <li><a href="#question-5" id="toc-question-5" class="nav-link" data-scroll-target="#question-5">Question 5</a></li>
  <li><a href="#question-6" id="toc-question-6" class="nav-link" data-scroll-target="#question-6">Question 6</a></li>
  <li><a href="#question-7" id="toc-question-7" class="nav-link" data-scroll-target="#question-7">Question 7</a></li>
  </ul></li>
  <li><a href="#bio312-final-exam-2021-2022" id="toc-bio312-final-exam-2021-2022" class="nav-link" data-scroll-target="#bio312-final-exam-2021-2022">BIO312 final exam (2021-2022)</a>
  <ul class="collapse">
  <li><a href="#question-1-1" id="toc-question-1-1" class="nav-link" data-scroll-target="#question-1-1">Question 1</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#sessioninfo" id="toc-sessioninfo" class="nav-link" data-scroll-target="#sessioninfo"><span style="color:green">SessionInfo</span></a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><strong>BIO312 Final Exam List</strong></h1>
<p class="subtitle lead"><em>Protein Structure and Function</em></p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p><a href="https://github.com/cupoftea0315" style="color:blue">TC-tea</a> </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">2024.06.01</p>
    </div>
  </div>
  
    
  </div>
  

</header>


<!-- Set viewport -->
<meta name="viewport" content="width=device-width, initial-scale=1">
<!-- Set styles for light and dark mode -->
<style>
    mark {
      background-color: yellow;
    }
    .dark-mode {
      background-color: black;
      color: white;
    }
    .light-code {
      color: black;
    }
    .dark-code {
      color: white;
    }
    .quarto-output pre:not([class*="language-"]) {
      background-color: transparent;
      color: inherit;
    }
    .quarto-output pre[class*="language-"] {
      background-color: #F5F5F5;
      color: black;
    }
    .dark-mode .quarto-output pre[class*="language-"] {
      background-color: #2B2B2B;
      color: white;
    }
    /* CSS for back-to-top button */
    #back-to-top {
      position: fixed;
      bottom: 10px;
      right: 22px;
      font-size: 22px;
      border-radius: 50%;
      width: 35px;
      height: 35px;
      text-decoration: none;
    }
    /* CSS for light/dark mode toggle button */
    #myBtn {
      position: fixed;
      top: 20px;
      left: 20px;
      border: none;
      border-radius: 10px;
      padding: 8px;
      background-color: lightgray;
    }
    #myBtn:before {
      content: "â˜€";
    }
    .dark-mode #myBtn:before {
      content: "ðŸŒ™";
    }
  </style>


<!-- Button to toggle light/dark mode -->
<button id="myBtn" onclick="myFunction()">
</button>
<script>
    function myFunction() {
      var element = document.body;
      element.classList.toggle("dark-mode");
      var codeBlocks = document.querySelectorAll(".quarto-output pre[class*='language-']");
      codeBlocks.forEach(function(block) {
        block.classList.toggle("light-code");
        block.classList.toggle("dark-code");
      });
    }
  </script>
<!-- Button to scroll back to top of page -->
<a href="#" id="back-to-top" title="Back to top">ðŸš€</a>

<section id="tool" class="level1">
<h1><strong>Tool</strong></h1>
<ul>
<li><a href="https://openai.com/index/chatgpt"><strong>ChatGPT</strong></a><br></li>
<li><a href="https://kimi.moonshot.cn/"><strong>Kimi AI</strong></a><br></li>
<li><a href="https://www.wolframalpha.com"><strong>Wolfram Alpha</strong></a><br></li>
<li><a href="https://core.xjtlu.edu.cn"><strong>Learning Mall</strong></a><br></li>
<li><a href="https://en.wikipedia.org/wiki/Main_Page"><strong>Wikipedia</strong></a></li>
<li><a href="https://jupyter.org/try"><strong>Jupyter Notebook</strong></a></li>
<li><a href="https://web.expasy.org/peptide_mass/"><strong>Peptide mass calculator</strong></a></li>
<li><a href="https://www.deepl.com/translator"><strong>DeepL</strong></a></li>
</ul>
</section>
<section id="courseware" class="level1">
<h1><strong>Courseware</strong></h1>
<section id="lecture" class="level2">
<h2 class="anchored" data-anchor-id="lecture">Lecture</h2>
<ul>
<li><a href="courseware/Lecture%201.pdf"><strong>Lecture 1</strong> (characterization of proteins)</a></li>
<li><a href="courseware/Lecture%202-1.pdf"><strong>Lecture 2.1</strong> (identification of proteins in complex mixtures <sub>a mass spectrometry approach</sub>)</a></li>
<li><a href="courseware/Lecture%202-2.pdf"><strong>Lecture 2.2</strong> (identification of proteins in complex mixtures <sub>a mass spectrometry approach</sub>)</a></li>
<li><a href="courseware/Lecture%203.pdf"><strong>Lecture 3</strong> (bottom-up proteomics and top-down proteomics)</a></li>
<li><a href="courseware/Lecture%204.pdf"><strong>Lecture 4</strong> (quantitative mass spectrometry-based proteomics)</a></li>
<li><a href="courseware/Lecture%205-1.pdf"><strong>Lecture 5.1</strong> (protein post translational modifications)</a></li>
<li><a href="courseware/Lecture%205-2.pdf"><strong>Lecture 5.2</strong> (protein post translational modifications)</a></li>
<li><a href="courseware/Lecture%206-1.pdf"><strong>Lecture 6</strong> (targeted proteomics)</a></li>
<li><a href="courseware/Lecture%207+8.pdf"><strong>Lecture 7+8</strong> (protein folding and unfolding)</a></li>
<li><a href="courseware/Lecture%209.pdf"><strong>Lecture 9.1</strong> (X-ray crystallography)</a></li>
<li><a href="courseware/Lecture%209-2.pdf"><strong>Lecture 9.2</strong> (X-ray crystallography)</a></li>
<li><a href="courseware/Lecture%2010-1.pdf"><strong>Lecture 10.1</strong> (nuclear magnetic resonance (NMR) spectroscopy)</a></li>
<li><a href="courseware/Lecture%2010-311.pdf"><strong>Lecture 10.2</strong> (nuclear magnetic resonance (NMR) spectroscopy)</a></li>
<li><a href="courseware/Lecture%2012.pdf"><strong>Lecture 12</strong> (protein modeling)</a></li>
<li><a href="courseware/Lecture%2013.pdf"><strong>Lecture 13</strong> (relating structure to function)</a></li>
<li><a href="courseware/Review.pdf"><strong>Review</strong></a></li>
</ul>
</section>
<section id="exercise" class="level2">
<h2 class="anchored" data-anchor-id="exercise">Exercise</h2>
<ul>
<li><a href="courseware/CW1.pdf"><strong>Coursework 1</strong></a></li>
<li><a href="courseware/CW2.pdf"><strong>Coursework 2</strong></a></li>
<li><a href="courseware/Tutorial%201.pdf"><strong>Tutorial 1</strong></a></li>
<li><a href="courseware/Tutorial%203.pdf"><strong>Tutorial 3</strong></a></li>
<li><a href="courseware/Tutorial%204.pdf"><strong>Tutorial 4</strong></a></li>
<li><a href="courseware/Tutorial%207+8.pdf"><strong>Tutorial 7+8</strong></a></li>
<li><a href="courseware/Tutorial%2010.pdf"><strong>Tutorial 10</strong></a></li>
<li><a href="courseware/Tutorial%20of%20CWs.pdf"><strong>Tutorial of CWs</strong></a></li>
</ul>
</section>
</section>
<section id="code" class="level1">
<h1><strong>Code</strong></h1>
<section id="de-novo-for-mz-plot" class="level2">
<h2 class="anchored" data-anchor-id="de-novo-for-mz-plot">De Novo for M/Z plot</h2>
<details>
<summary>
Example of M/Z analysis
</summary>
<img src="images/yuxuan_tian.jpg" class="img-fluid"> <img src="images/amino%20acids%20mass%20table.jpg" class="img-fluid">
</details>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># De Novo sequencing</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> find_matching_indices(data, values):</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>    indices <span class="op">=</span> []</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> i <span class="kw">in</span> <span class="bu">range</span>(<span class="bu">len</span>(data)):</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> j <span class="kw">in</span> <span class="bu">range</span>(i <span class="op">+</span> <span class="dv">1</span>, <span class="bu">len</span>(data)):</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>            diff <span class="op">=</span> <span class="bu">abs</span>(data[i] <span class="op">-</span> data[j])</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> <span class="bu">any</span>(<span class="bu">abs</span>(diff <span class="op">-</span> value) <span class="op">&lt;</span> <span class="fl">5e-1</span> <span class="cf">for</span> value <span class="kw">in</span> values):  <span class="co"># determine the difference</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>                indices.append((i <span class="op">+</span> <span class="dv">1</span>, j <span class="op">+</span> <span class="dv">1</span>, diff))</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> indices</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co"># M/Z peak data</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>data <span class="op">=</span> [<span class="fl">175.12</span>, <span class="fl">262.12</span>, <span class="fl">274.19</span>, <span class="fl">375.21</span>, <span class="fl">446.23</span>, <span class="fl">487.32</span>, <span class="fl">602.32</span>, <span class="fl">730.43</span>, <span class="fl">859.45</span>, <span class="fl">875.42</span>, <span class="fl">916.49</span>, <span class="fl">987.52</span>, <span class="fl">1100.61</span>,</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>        <span class="fl">1247.69</span>, <span class="fl">1360.9</span>]</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Mass of amino acids</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>values <span class="op">=</span> [<span class="fl">57.02147</span>, <span class="fl">71.03712</span>, <span class="fl">87.03203</span>, <span class="fl">97.05277</span>, <span class="fl">99.06842</span>, <span class="fl">101.04768</span>, <span class="fl">103.00919</span>, <span class="fl">113.08407</span>, <span class="fl">114.04293</span>, <span class="fl">115.02695</span>,</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>          <span class="fl">128.05858</span>, <span class="fl">128.09497</span>, <span class="fl">129.04260</span>, <span class="fl">131.04049</span>, <span class="fl">137.05891</span>, <span class="fl">147.06842</span>, <span class="fl">156.10112</span>, <span class="fl">163.06333</span>, <span class="fl">186.07932</span>]</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the difference and find the same position as the given value</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>matching_indices <span class="op">=</span> find_matching_indices(data, values)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Results</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> index <span class="kw">in</span> matching_indices:</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>    <span class="bu">print</span>(<span class="ss">f"#</span><span class="sc">{</span>index[<span class="dv">0</span>]<span class="sc">}</span><span class="ss"> - #</span><span class="sc">{</span>index[<span class="dv">1</span>]<span class="sc">}</span><span class="ss"> = </span><span class="sc">{</span>index[<span class="dv">2</span>]<span class="sc">:.2f}</span><span class="ss">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>#1 - #2 = 87.00
#1 - #3 = 99.07
#2 - #4 = 113.09
#3 - #4 = 101.02
#4 - #5 = 71.02
#5 - #7 = 156.09
#6 - #7 = 115.00
#7 - #8 = 128.11
#8 - #9 = 129.02
#8 - #11 = 186.06
#9 - #11 = 57.04
#9 - #12 = 128.07
#11 - #12 = 71.03
#12 - #13 = 113.09
#13 - #14 = 147.08
#14 - #15 = 113.21</code></pre>
</div>
</div>
</section>
<section id="molecular-weight-of-mass-spectrum" class="level2">
<h2 class="anchored" data-anchor-id="molecular-weight-of-mass-spectrum">Molecular weight of mass spectrum</h2>
<details>
<summary>
Example of mass spectrum
</summary>
<img src="images/mass%20spectrum.png" class="img-fluid">
</details>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the charge (19+ 18+ 17+ 16+ 15+)</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>z18 <span class="op">=</span> (<span class="fl">3212.1</span><span class="op">-</span><span class="fl">1.0078</span>)<span class="op">/</span>(<span class="fl">3390.4</span><span class="op">-</span><span class="fl">3212.1</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>z18</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>18.0094907459338</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Mass of protein A</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>mA <span class="op">=</span> ((<span class="dv">12</span><span class="op">*</span><span class="fl">2128.1</span><span class="op">-</span><span class="dv">12</span>)<span class="op">+</span>(<span class="dv">11</span><span class="op">*</span><span class="fl">2321.5</span><span class="op">-</span><span class="dv">11</span>)<span class="op">+</span>(<span class="dv">10</span><span class="op">*</span><span class="fl">2553.5</span><span class="op">-</span><span class="dv">10</span>)<span class="op">+</span>(<span class="dv">9</span><span class="op">*</span><span class="fl">2837.1</span><span class="op">-</span><span class="dv">9</span>))<span class="op">/</span><span class="dv">4</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>mA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>25525.149999999998</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Mass of AB complex</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>mAB <span class="op">=</span> ((<span class="dv">19</span><span class="op">*</span><span class="fl">3212.1</span><span class="op">-</span><span class="dv">19</span>)<span class="op">+</span>(<span class="dv">18</span><span class="op">*</span><span class="fl">3390.4</span><span class="op">-</span><span class="dv">18</span>)<span class="op">+</span>(<span class="dv">17</span><span class="op">*</span><span class="fl">3589.8</span><span class="op">-</span><span class="dv">17</span>)<span class="op">+</span>(<span class="dv">16</span><span class="op">*</span><span class="fl">3814.1</span><span class="op">-</span><span class="dv">16</span>)<span class="op">+</span>(<span class="dv">15</span><span class="op">*</span><span class="fl">4068.3</span><span class="op">-</span><span class="dv">15</span>))<span class="op">/</span><span class="dv">5</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>mAB</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>61009.76000000001</code></pre>
</div>
</div>
</section>
</section>
<section id="past-exam-paper" class="level1">
<h1><strong>Past exam paper</strong></h1>
<section id="bio312-final-exam-2022-2023" class="level2">
<h2 class="anchored" data-anchor-id="bio312-final-exam-2022-2023">BIO312 final exam (2022-2023)</h2>
<section id="question-1" class="level3">
<h3 class="anchored" data-anchor-id="question-1">Question 1</h3>
<details>
<summary>
<strong>Q1</strong>: 2D-DIGE gel &amp; MS spectrometry
</summary>
<img src="images/BIO312%20final%20exam%20q1%20(2022-2023).jpg" class="img-fluid">
</details>
<p>Solution:</p>
<ol type="1">
<li>They have different PH levels, and Mass Spectrometry technology can be used to separate these three proteins.</li>
<li>Steps:
<ul>
<li>Excise the protein spots from the gel.</li>
<li>Digest the proteins into peptides using trypsin.</li>
<li>Extract and purify the peptides from the gel pieces.</li>
<li>Ionize the peptides using techniques like Electrospray Ionization (ESI) or Matrix-Assisted Laser Desorption/Ionization (MALDI).</li>
<li>Analyze the peptides using a Mass Spectrometer to obtain their mass-to-charge ratios (m/z).</li>
<li>Select the peptides of interest for fragmentation.</li>
<li>Fragmentize the selected peptides using Collision-Induced Dissociation (CID) or Higher Energy Collisional Dissociation (HCD).</li>
<li>Analyze the resulting fragment ions to identify characteristic signs of phosphorylation, such as loss of the phosphate group (e.g., -H3PO4).</li>
<li>Use bioinformatics tools to match the peptide sequences and confirm the presence of phosphorylated tyrosine residues.</li>
</ul></li>
</ol>
</section>
<section id="question-2" class="level3">
<h3 class="anchored" data-anchor-id="question-2">Question 2</h3>
<details>
<summary>
<strong>Q2</strong>: MS spectrum
</summary>
<img src="images/BIO312%20final%20exam%20q2%20(2022-2023).jpg" class="img-fluid">
</details>
<p>Solution:</p>
<ol type="1">
<li>(681.45*2)-2=1360.9</li>
<li>D+P | I+V | L+V</li>
<li>Same as the CW1 q1, <span style="color:red">N F I/L A G E Q/K D X X V R</span> (XX could be NV, VN, RG, GR).</li>
</ol>
</section>
<section id="question-3" class="level3">
<h3 class="anchored" data-anchor-id="question-3">Question 3</h3>
<details>
<summary>
<strong>Q3</strong>: MS and MS/MS spectra
</summary>
<img src="images/BIO312%20final%20exam%20q3%20(2022-2023).jpg" class="img-fluid">
</details>
<p>Solution:</p>
<ol type="1">
<li>m/z=693.1 &amp; m/z=1422.0</li>
<li>Steps:
<ul>
<li>Perform Spike-In Experiments: Introduce a known quantity of the NHL1 peptide into the sample to confirm that the selected transitions are detectable and accurately quantify the peptide.</li>
<li>Optimize SRM Parameters: Adjust the collision energy and other SRM settings to ensure the selected transitions produce the most intense and specific signals.</li>
<li>Analyse MS/MS Spectra: Confirm that the selected transitions match the expected fragment ions by comparing the experimental spectra to theoretical predictions or database entries.</li>
<li>Evaluate Reproducibility: Assess the consistency of the transitions across multiple runs to ensure they are reliable for quantification purposes.</li>
<li>Check for Cross-Contamination: Ensure that the transitions are unique to the NHL1 peptide and not shared with other peptides that could lead to false positives.</li>
</ul></li>
<li>Triple quadrupole mass spectrometer is typically used for SRM assays due to its ability to selectively monitor specific peptide transitions.</li>
<li>List:
<ul>
<li>Data-Dependent Acquisition (DDA): To identify and quantify peptides without prior knowledge of the sample composition.</li>
<li>Data-Independent Acquisition (DIA): An unbiased approach that captures a wide range of peptides for comprehensive analysis.</li>
<li>Isobaric Tagging: Such as tandem mass tag (TMT) or isobaric tags for relative and absolute quantitation (iTRAQ), for multiplexed comparative proteomics.</li>
<li>Western Blot: A gel-based method to confirm protein expression levels.</li>
<li>Enzyme-Linked Immunosorbent Assay (ELISA): For quantitative analysis of specific proteins using antibodies.</li>
<li>Peptide Enrichment Techniques: Such as immunoprecipitation or affinity chromatography, to selectively enrich and quantify target peptides.</li>
</ul></li>
</ol>
</section>
<section id="question-4" class="level3">
<h3 class="anchored" data-anchor-id="question-4">Question 4</h3>
<details>
<summary>
<strong>Q4</strong>: X-ray Crystallography &amp; NMR Spectroscopy
</summary>
<img src="images/BIO312%20final%20exam%20q4%20(2022-2023).jpg" class="img-fluid">
</details>
<p>Solution:</p>
<ul>
<li>X-ray Crystallography Limitations:
<ul>
<li>Crystal Dependency: Requires high-quality crystals that are not always achievable.</li>
<li>Radiation Sensitivity: Susceptible to radiation damage affecting the protein structure.</li>
<li>Static Structure: Captures a single conformation, missing dynamic aspects.</li>
</ul></li>
<li>NMR Spectroscopy Limitations:
<ul>
<li>Size Restriction: Limited to smaller proteins due to spectral complexity.</li>
<li>Sensitivity Issues: Requires larger sample amounts and longer times.</li>
<li>Structural Detail: Provides less detailed structural information compared to X-ray crystallography.</li>
</ul></li>
</ul>
</section>
<section id="question-5" class="level3">
<h3 class="anchored" data-anchor-id="question-5">Question 5</h3>
<details>
<summary>
<strong>Q5</strong>: CD spectrum
</summary>
<img src="images/BIO312%20final%20exam%20q5%20(2022-2023).jpg" class="img-fluid">
</details>
<p>Solution:</p>
<ol type="1">
<li><strong>A</strong>, 1BRR protein is basically composed of alpha-helix and beta-sheets, this structure generally present a negative peak around 218 nm and a positive peak around 195 nm in the CD spectrum.</li>
<li><strong>E</strong>, Urea is a denaturant that disrupts the non-covalent interactions that stabilize protein structure. If the protein were treated with 8 M urea, it would expect to cause a loss of the characteristic secondary structure peaks, shifting towards a spectrum, which is typically a weak and featureless spectrum with a minimum around 200 nm.</li>
<li>In the absence of ligands, mutant would have fewer alpha-helix and beta-sheets than wild type. After the addition of ligands, both alpha-helix and beta-sheet of wild type would drastically decrease, but mutant beta-sheet would be more affected at this time.</li>
</ol>
</section>
<section id="question-6" class="level3">
<h3 class="anchored" data-anchor-id="question-6">Question 6</h3>
<details>
<summary>
<strong>Q6</strong>: Ramachandran plot
</summary>
<img src="images/BIO312%20final%20exam%20q6%20(2022-2023).jpg" class="img-fluid">
</details>
<p>Solution:</p>
<ul>
<li>The Ramachandran plot is essential as it evaluates the steriochemical quality of a protein structure by plotting the phi (Ï†) and psi (Ïˆ) backbone dihedral angles. It helps identify if the proteinâ€™s backbone conforms to allowed regions based on known protein geometries, ensuring the modelâ€™s accuracy.</li>
<li>It turns out that the enzyme in Plot B has a well-defined conformation and the crystallography data is of higher quality.
<ul>
<li>Plot A: The points are scattered and many are outside the favored red region, indicating potential issues with the protein modelâ€™s accuracy. This dispersion could be due to structural errors, poor data quality, or the presence of multiple conformations.</li>
<li>Plot B: The points are predominantly within the favored red region, suggesting a high-quality protein model with correct backbone geometry and fewer structural anomalies.</li>
</ul></li>
</ul>
</section>
<section id="question-7" class="level3">
<h3 class="anchored" data-anchor-id="question-7">Question 7</h3>
<details>
<summary>
<strong>Q7</strong>: Surface Plasmon Resonance (SPR)
</summary>
<img src="images/BIO312%20final%20exam%20q7%20(2022-2023).jpg" class="img-fluid">
</details>
<p>Solution:</p>
<ol type="1">
<li><span class="math display">\[K_i=\frac{koff}{kon}\]</span>
<ul>
<li>Association Rate (<span class="math inline">\(kon\)</span>): The initial slope of the binding phase is proportional to the association rate constant.</li>
<li>Dissociation Rate (<span class="math inline">\(koff\)</span>): The slope of the dissociation phase is related to the dissociation rate constant.</li>
<li>Equilibrium Binding Affinity (<span class="math inline">\(K_D\)</span> or <span class="math inline">\(K_i\)</span>): The affinity can be calculated using the ratio <span class="math inline">\(koff\)</span>/<span class="math inline">\(kon\)</span>.</li>
</ul></li>
<li><strong>Inhibitor C</strong>, the <span class="math inline">\(koff\)</span> is much more smaller than <span class="math inline">\(kon\)</span>, so <span class="math inline">\(K_i\)</span> can be 0.5 nM.</li>
<li><strong>Inhibitor B</strong>, the <span class="math inline">\(koff\)</span> is relatively high, so according to <span class="math inline">\(K_i=\frac{koff}{kon}\)</span> we can get the <span class="math inline">\(K_i\)</span> of B is relatively higher than others.</li>
<li>For chronic diseases, an inhibitor with a high affinity (low K value) is desirable to maintain therapeutic levels with less frequent dosing. <strong>Inhibitor C, with the lowest <span class="math inline">\(K\)</span> value (0.5 nM), would be the most suitable.</strong> It would likely provide a longer duration of action and potentially fewer side effects due to its high binding affinity, making it ideal for consistent management of conditions like epilepsy.</li>
</ol>
</section>
</section>
<section id="bio312-final-exam-2021-2022" class="level2">
<h2 class="anchored" data-anchor-id="bio312-final-exam-2021-2022">BIO312 final exam (2021-2022)</h2>
<section id="question-1-1" class="level3">
<h3 class="anchored" data-anchor-id="question-1-1">Question 1</h3>
<details>
<summary>
<strong>Q1</strong>:
</summary>
<img src="images/BIO312%20final%20exam%20q1%20(2021-2022).jpg" class="img-fluid">
</details>
<p>Solution:</p>
</section>
</section>
</section>
<section id="sessioninfo" class="level1">
<h1><span style="color:green">SessionInfo</span></h1>
<div class="cell" data-layout-align="center">
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.4.0 (2024-04-24 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=Chinese (Simplified)_China.utf8 
[2] LC_CTYPE=Chinese (Simplified)_China.utf8   
[3] LC_MONETARY=Chinese (Simplified)_China.utf8
[4] LC_NUMERIC=C                               
[5] LC_TIME=en_GB.UTF-8                        

time zone: Etc/GMT-8
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] digest_0.6.35     fastmap_1.1.1     xfun_0.43         Matrix_1.7-0     
 [5] lattice_0.22-6    reticulate_1.37.0 rappdirs_0.3.3    knitr_1.46       
 [9] htmltools_0.5.8.1 png_0.1-8         rmarkdown_2.26    cli_3.6.2        
[13] grid_4.4.0        withr_3.0.0       compiler_4.4.0    rprojroot_2.0.4  
[17] here_1.0.1        rstudioapi_0.16.0 tools_4.4.0       evaluate_0.23    
[21] Rcpp_1.0.12       yaml_2.3.8        rlang_1.1.3       jsonlite_1.8.8   
[25] htmlwidgets_1.6.4</code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>