---
title: "**BIO312 Final Exam List**"
subtitle: "*Protein Structure and Function*"
author:
  - "[TC-tea](https://github.com/cupoftea0315){style='color:blue'}"
date: "`r Sys.Date()`"
date-format: "YYYY.MM.DD"
output:
  quarto::quarto_document:
    default_output_format: html
    code:
      pandoc_args: ["-V", "lang=en"]
    df_print: paged
toc: true
toc-location: left
toc-depth: 4
fontsize: 12pt
fontfamily: Tahoma
theme: default
comments:
  hypothesis: true
---

<head>
  <!-- Set viewport -->
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <!-- Set styles for light and dark mode -->
  <style>
    mark {
      background-color: yellow;
    }
    .dark-mode {
      background-color: black;
      color: white;
    }
    .light-code {
      color: black;
    }
    .dark-code {
      color: white;
    }
    .quarto-output pre:not([class*="language-"]) {
      background-color: transparent;
      color: inherit;
    }
    .quarto-output pre[class*="language-"] {
      background-color: #F5F5F5;
      color: black;
    }
    .dark-mode .quarto-output pre[class*="language-"] {
      background-color: #2B2B2B;
      color: white;
    }
    /* CSS for back-to-top button */
    #back-to-top {
      position: fixed;
      bottom: 10px;
      right: 22px;
      font-size: 22px;
      border-radius: 50%;
      width: 35px;
      height: 35px;
      text-decoration: none;
    }
    /* CSS for light/dark mode toggle button */
    #myBtn {
      position: fixed;
      top: 20px;
      left: 20px;
      border: none;
      border-radius: 10px;
      padding: 8px;
      background-color: lightgray;
    }
    #myBtn:before {
      content: "‚òÄ";
    }
    .dark-mode #myBtn:before {
      content: "üåô";
    }
  </style>
</head>

<body>
  <!-- Button to toggle light/dark mode -->
  <button id="myBtn" onclick="myFunction()"></button>
  <script>
    function myFunction() {
      var element = document.body;
      element.classList.toggle("dark-mode");
      var codeBlocks = document.querySelectorAll(".quarto-output pre[class*='language-']");
      codeBlocks.forEach(function(block) {
        block.classList.toggle("light-code");
        block.classList.toggle("dark-code");
      });
    }
  </script>
  <!-- Button to scroll back to top of page -->
<a href="#" id="back-to-top" title="Back to top">üöÄ</a>
</body>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      message = FALSE,
                      warning = FALSE,
                      fig.align = "center")
Sys.setlocale("LC_TIME", "en_GB.UTF-8")
```

# **Tool**
- [**ChatGPT**](https://openai.com/index/chatgpt)<br>
- [**Kimi AI**](https://kimi.moonshot.cn/)<br>
- [**Wolfram Alpha**](https://www.wolframalpha.com)<br>
- [**Learning Mall**](https://core.xjtlu.edu.cn)<br>
- [**Wikipedia**](https://en.wikipedia.org/wiki/Main_Page)
- [**Jupyter Notebook**](https://jupyter.org/try)
- [**Peptide mass calculator**](https://web.expasy.org/peptide_mass/)
- [**DeepL**](https://www.deepl.com/translator)

# **Courseware**
## Lecture
- [**Lecture 1** (characterization of proteins)](courseware/Lecture 1.pdf)
- [**Lecture 2.1** (identification of proteins in complex mixtures <sub>a mass spectrometry approach</sub>)](courseware/Lecture 2-1.pdf)
- [**Lecture 2.2** (identification of proteins in complex mixtures <sub>a mass spectrometry approach</sub>)](courseware/Lecture 2-2.pdf)
- [**Lecture 3** (bottom-up proteomics and top-down proteomics)](courseware/Lecture 3.pdf)
- [**Lecture 4** (quantitative mass spectrometry-based proteomics)](courseware/Lecture 4.pdf)
- [**Lecture 5.1** (protein post translational modifications)](courseware/Lecture 5-1.pdf)
- [**Lecture 5.2** (protein post translational modifications)](courseware/Lecture 5-2.pdf)
- [**Lecture 6** (targeted proteomics)](courseware/Lecture 6-1.pdf)
- [**Lecture 7+8** (protein folding and unfolding)](courseware/Lecture 7+8.pdf)
- [**Lecture 9.1** (X-ray crystallography)](courseware/Lecture 9.pdf)
- [**Lecture 9.2** (X-ray crystallography)](courseware/Lecture 9-2.pdf)
- [**Lecture 10.1** (nuclear magnetic resonance (NMR) spectroscopy)](courseware/Lecture 10-1.pdf)
- [**Lecture 10.2** (nuclear magnetic resonance (NMR) spectroscopy)](courseware/Lecture 10-311.pdf)
- [**Lecture 12** (protein modeling)](courseware/Lecture 12.pdf)
- [**Lecture 13** (relating structure to function)](courseware/Lecture 13.pdf)
- [**Review**](courseware/Review.pdf)

## Exercise
- [**Coursework 1**](courseware/CW1.pdf)
- [**Coursework 2**](courseware/CW2.pdf)
- [**Tutorial 1** (protein separation)](courseware/Tutorial 1.pdf)
- [**Tutorial 3** (MS spectrum interpretation)](courseware/Tutorial 3.pdf)
- [**Tutorial 4** (quantification methods in proteomics)](courseware/Tutorial 4.pdf)
- [**Tutorial 7+8** (Protein folding)](courseware/Tutorial 7+8.pdf)
- [**Tutorial 10** (nuclear magnetic resonance (NMR) spectroscopy)](courseware/Tutorial 10.pdf)
- [**Tutorial of courseworks**](courseware/Tutorial of CWs.pdf)

# **Code**
## De Novo for M/Z plot
<details>
  <summary>Example of M/Z analysis</summary>
  ![](images/yuxuan_tian.jpg)
  ![](images/amino acids mass table.jpg)
</details>
```{python}
# De Novo sequencing
def find_matching_indices(data, values):
    indices = []
    for i in range(len(data)):
        for j in range(i + 1, len(data)):
            diff = abs(data[i] - data[j])
            if any(abs(diff - value) < 5e-1 for value in values):  # determine the difference
                indices.append((i + 1, j + 1, diff))
    return indices
  
# M/Z peak data
data = [175.12, 262.12, 274.19, 375.21, 446.23, 487.32, 602.32, 730.43, 859.45, 875.42, 916.49, 987.52, 1100.61,
        1247.69, 1360.9]
        
# Mass of amino acids
values = [57.02147, 71.03712, 87.03203, 97.05277, 99.06842, 101.04768, 103.00919, 113.08407, 114.04293, 115.02695,
          128.05858, 128.09497, 129.04260, 131.04049, 137.05891, 147.06842, 156.10112, 163.06333, 186.07932]
          
# Calculate the difference and find the same position as the given value
matching_indices = find_matching_indices(data, values)

# Results
for index in matching_indices:
    print(f"#{index[0]} - #{index[1]} = {index[2]:.2f}")
```

## Molecular weight of mass spectrum
<details>
  <summary>Example of mass spectrum</summary>
  ![](images/mass spectrum.png)
</details>
```{python}
# Calculate the charge (19+ 18+ 17+ 16+ 15+)
z18 = (3212.1-1.0078)/(3390.4-3212.1)
z18
# Mass of protein A
mA = ((12*2128.1-12)+(11*2321.5-11)+(10*2553.5-10)+(9*2837.1-9))/4
mA
# Mass of AB complex
mAB = ((19*3212.1-19)+(18*3390.4-18)+(17*3589.8-17)+(16*3814.1-16)+(15*4068.3-15))/5
mAB
```

# **Past exam paper**
## BIO312 final exam (2022-2023)
### Question 1
<details>
  <summary>**Q1**: 2D-DIGE gel & MS spectrometry</summary>
  ![](images/BIO312 final exam q1 (2022-2023).jpg)
</details>
Solution:

1. They have different PH levels, and Mass Spectrometry technology can be used to separate these three proteins.
2. Steps:
   - Excise the protein spots from the gel.
   - Digest the proteins into peptides using trypsin.
   - Extract and purify the peptides from the gel pieces.
   - Ionize the peptides using techniques like Electrospray Ionization (ESI) or Matrix-Assisted Laser Desorption/Ionization (MALDI).
   - Analyze the peptides using a Mass Spectrometer to obtain their mass-to-charge ratios (m/z).
   - Select the peptides of interest for fragmentation.
   - Fragmentize the selected peptides using Collision-Induced Dissociation (CID) or Higher Energy Collisional Dissociation (HCD).
   - Analyze the resulting fragment ions to identify characteristic signs of phosphorylation, such as loss of the phosphate group (e.g., -H3PO4).
   - Use bioinformatics tools to match the peptide sequences and confirm the presence of phosphorylated tyrosine residues.

### Question 2
<details>
  <summary>**Q2**: MS spectrum</summary>
  ![](images/BIO312 final exam q2 (2022-2023).jpg)
</details>
Solution:

1. (681.45*2)-2=1360.9
2. D+P | I+V | L+V
3. Same as the CW1 q1, [N F I/L A G E Q/K D X X V R]{style="color:red"} (XX could be NV, VN, RG, GR).

### Question 3
<details>
  <summary>**Q3**: MS and MS/MS spectra</summary>
  ![](images/BIO312 final exam q3 (2022-2023).jpg)
</details>
Solution:

1. m/z=693.1 & m/z=1422.0
2. Steps:
   - Perform Spike-In Experiments: Introduce a known quantity of the NHL1 peptide into the sample to confirm that the selected transitions are detectable and accurately quantify the peptide.
   - Optimize SRM Parameters: Adjust the collision energy and other SRM settings to ensure the selected transitions produce the most intense and specific signals.
   - Analyse MS/MS Spectra: Confirm that the selected transitions match the expected fragment ions by comparing the experimental spectra to theoretical predictions or database entries.
   - Evaluate Reproducibility: Assess the consistency of the transitions across multiple runs to ensure they are reliable for quantification purposes.
   - Check for Cross-Contamination: Ensure that the transitions are unique to the NHL1 peptide and not shared with other peptides that could lead to false positives.
3. Triple quadrupole mass spectrometer is typically used for SRM assays due to its ability to selectively monitor specific peptide transitions.
4. List:
   - Data-Dependent Acquisition (DDA): To identify and quantify peptides without prior knowledge of the sample composition.
   - Data-Independent Acquisition (DIA): An unbiased approach that captures a wide range of peptides for comprehensive analysis.
   - Isobaric Tagging: Such as tandem mass tag (TMT) or isobaric tags for relative and absolute quantitation (iTRAQ), for multiplexed comparative proteomics.
   - Western Blot: A gel-based method to confirm protein expression levels.
   - Enzyme-Linked Immunosorbent Assay (ELISA): For quantitative analysis of specific proteins using antibodies.
   - Peptide Enrichment Techniques: Such as immunoprecipitation or affinity chromatography, to selectively enrich and quantify target peptides.

### Question 4
<details>
  <summary>**Q4**: X-ray Crystallography & NMR Spectroscopy</summary>
  ![](images/BIO312 final exam q4 (2022-2023).jpg)
</details>
Solution:

- X-ray Crystallography Limitations:
   - Crystal Dependency: Requires high-quality crystals that are not always achievable.
   - Radiation Sensitivity: Susceptible to radiation damage affecting the protein structure.
   - Static Structure: Captures a single conformation, missing dynamic aspects.
- NMR Spectroscopy Limitations:
   - Size Restriction: Limited to smaller proteins due to spectral complexity.
   - Sensitivity Issues: Requires larger sample amounts and longer times.
   - Structural Detail: Provides less detailed structural information compared to X-ray crystallography.

### Question 5
<details>
  <summary>**Q5**: CD spectrum</summary>
  ![](images/BIO312 final exam q5 (2022-2023).jpg)
</details>
Solution:

1. **A**, 1BRR protein is basically composed of alpha-helix and beta-sheets, this structure generally present a negative peak around 218 nm and a positive peak around 195 nm in the CD spectrum.
2. **E**, Urea is a denaturant that disrupts the non-covalent interactions that stabilize protein structure. If the protein were treated with 8 M urea, it would expect to cause a loss of the characteristic secondary structure peaks, shifting towards a spectrum, which is typically a weak and featureless spectrum with a minimum around 200 nm.
3. In the absence of ligands, mutant would have fewer alpha-helix and beta-sheets than wild type. After the addition of ligands, both alpha-helix and beta-sheet of wild type would drastically decrease, but mutant beta-sheet would be more affected at this time.

### Question 6
<details>
  <summary>**Q6**: Ramachandran plot</summary>
  ![](images/BIO312 final exam q6 (2022-2023).jpg)
</details>
Solution:

- The Ramachandran plot is essential as it evaluates the steriochemical quality of a protein structure by plotting the phi (œÜ) and psi (œà) backbone dihedral angles. It helps identify if the protein's backbone conforms to allowed regions based on known protein geometries, ensuring the model's accuracy.
- It turns out that the enzyme in Plot B has a well-defined conformation and the crystallography data is of higher quality.
   - Plot A: The points are scattered and many are outside the favored red region, indicating potential issues with the protein model's accuracy. This dispersion could be due to structural errors, poor data quality, or the presence of multiple conformations.
   - Plot B: The points are predominantly within the favored red region, suggesting a high-quality protein model with correct backbone geometry and fewer structural anomalies.

### Question 7
<details>
  <summary>**Q7**: Surface Plasmon Resonance (SPR)</summary>
  ![](images/BIO312 final exam q7 (2022-2023).jpg)
</details>
Solution:

1. $$K_i=\frac{koff}{kon}$$
   - Association Rate ($kon$): The initial slope of the binding phase is proportional to the association rate constant.
   - Dissociation Rate ($koff$): The slope of the dissociation phase is related to the dissociation rate constant.
   - Equilibrium Binding Affinity ($K_D$ or $K_i$): The affinity can be calculated using the ratio $koff$/$kon$.
2. **Inhibitor C**, the $koff$ is much more smaller than $kon$, so $K_i$ can be 0.5 nM.
3. **Inhibitor B**, the $koff$ is relatively high, so according to $K_i=\frac{koff}{kon}$ we can get the $K_i$ of B is relatively higher than others.
4. For chronic diseases, an inhibitor with a high affinity (low K value) is desirable to maintain therapeutic levels with less frequent dosing. **Inhibitor C, with the lowest $K$ value (0.5 nM), would be the most suitable.** It would likely provide a longer duration of action and potentially fewer side effects due to its high binding affinity, making it ideal for consistent management of conditions like epilepsy.

## BIO312 final exam (2021-2022)
### Question 1
<details>
  <summary>**Q1**: MS1 spectrum & tandem mass spectrum</summary>
  ![](images/BIO312 final exam q1 (2021-2022) (1).jpg)
  ![](images/BIO312 final exam q1 (2021-2022) (2).jpg)
</details>
Solution:

1. Mass of the modified peptide: 690.87 Da<br>Charge of this peptide: +2
2. 
3. De Novo sequencing: D V A I/L A A A I/L A V

### Question 2
<details>
  <summary>**Q2**:</summary>
  ![](images/BIO312 final exam q2 (2021-2022).jpg)
</details>
Solution:

1. To address Researcher A's question efficiently, we can follow these steps:
   - Size Exclusion Chromatography (SEC): Since Researcher A has already performed SEC and found the enzyme activity concentrated in the 30-50 kDa fraction, this step has effectively narrowed down the protein candidates.
   - One-Dimensional Gel Electrophoresis (1D-PAGE): Next, apply 1D-PAGE to the 30-50 kDa fraction to further separate and purify the proteins. This will allow visualization of distinct protein bands, from which the band(s) of interest can be excised.
   - Matrix-Assisted Laser Desorption/Ionization Time-of-Flight Mass Spectrometry (MALDI-TOF MS): Use the MALDI-TOF MS in reflectron mode to determine the exact mass of the proteins in the excised band(s). This method is highly accurate for protein mass determination.
   - Tandem Quadrupole Time-of-Flight Mass Spectrometry (tandem Q-TOF MS): If additional information on protein identity and sequence is required, tandem Q-TOF MS can be employed. This instrument can provide fragmentation data that, when combined with a protein database, can lead to the identification of the protein.
2. To assist Researcher B in identifying the synthetic peptide, we can use the following approach:
   - Reverse-Phase High-Performance Liquid Chromatography (RP-HPLC): Researcher B has already utilized RP-HPLC to separate the peptide mixture. This step is based on hydrophobicity and has resulted in 56 fractions.
   - Nanoflow HPLC System: To further purify and analyze the fractions, use the nanoflow HPLC system. This will increase the resolution and purity of the peptide, which is especially important given the complexity of the mixture.
   - Tandem Quadrupole Time-of-Flight Mass Spectrometry (tandem Q-TOF MS): Couple the nanoflow HPLC directly to the tandem Q-TOF MS for online analysis. This setup allows for the real-time detection and identification of peptides as they elute from the HPLC column. The tandem MS capability will provide sequence information, which is crucial for identifying the synthetic peptide.
   - MALDI-TOF MS: For the fractions that are suspected to contain the synthetic peptide based on the HPLC-MS/MS data, use MALDI-TOF MS to rapidly confirm the presence of the peptide by matching its mass to the known monoisotopic mass of 2689.2 Da.

### Question 3
<details>
  <summary>**Q3**:</summary>
  ![](images/BIO312 final exam q3 (2021-2022).jpg)
</details>
Solution:

1. Serine (Ser, S), Threonine (Thr, T), and Tyrosine (Tyr, Y) are the primary amino acids that can be phosphorylated.
2. To enrich phosphopeptides:
   - Use immobilized metal affinity chromatography (IMAC) with metal ions like Fe¬≥‚Å∫ or Ga¬≥‚Å∫ to selectively bind phosphorylated peptides.
   - Employ cation exchange chromatography to separate phosphorylated peptides based on their increased negative charge.
   - Apply solid-phase extraction (SPE) to purify and concentrate phosphopeptides from the digest.
3. Phosphopeptides often exhibit lower fragmentation efficiency in CID spectra due to the stabilizing effect of the phosphate group. This results in fewer and less intense fragment ions, making it challenging to pinpoint the exact site of phosphorylation within a peptide. The reduced fragmentation can lead to ambiguity in assigning the modification to a specific amino acid residue.

### Question 4a
<details>
  <summary>**Q4a**:</summary>
  ![](images/BIO312 final exam q4a (2021-2022).jpg)
</details>
Solution: 

1. The observation likely indicates post-translational modifications (PTMs) of a protein in cancerous tissue, altering its molecular weight (MW) and isoelectric point (pI). Verification can be done through:
   - Mass Spectrometry: To identify the protein and detect PTMs.
   - Western Blot: Using specific antibodies to confirm protein identity.
2. Biomarker Discovery Workflow:
   - Sample Collection: Obtain normal and cancerous kidney tissues.
   - Protein Extraction: Prepare protein samples for analysis.
   - 2D DIGE: Separate proteins based on pI and MW.
   - Image Analysis: Identify differentially expressed proteins.
   - Protein Identification: Excise and digest protein spots for mass spectrometry.
   - Bioinformatics: Analyze data to predict protein functions.
   - Validation: Use Western blot or ELISA to confirm expression.
   - Functional Analysis: Investigate biological roles of proteins.
   - Clinical Correlation: Relate protein expression to patient data.
   - Validation Studies: Validate potential biomarkers in larger cohorts.

### Question 4b
<details>
  <summary>**Q4b**:</summary>
  ![](images/BIO312 final exam q4b (2021-2022).jpg)
</details>
Solution:

1. Fractionation Method:
   - High-Resolution Gel Electrophoresis [Protein]: Such as 2D gel electrophoresis or Blue Native PAGE, to separate proteins based on their isoelectric point and molecular weight.
   - Reversed-Phase Liquid Chromatography (RPLC) [Peptide]: To separate peptides based on their hydrophobicity.
   - Hydrophilic Interaction Liquid Chromatography (HILIC) [Peptide]: To separate peptides based on their hydrophilicity.
2. Workflow/Strategy for Absolute Quantification of AMPD Enzyme:
   - Standard Preparation: Synthesize or obtain a set of isotopically labeled (heavy) AMPD peptides to serve as internal standards.
   - Sample Preparation: Homogenize liver tissue from both control and high fructose diet-fed rats, followed by protein extraction.
   - Protein Digestion: Digest the extracted proteins using a protease like trypsin to generate peptides.
   - Peptide Fractionation:
      - Use RPLC to fractionate peptides based on hydrophobicity.
      - Employ HILIC to further separate hydrophilic peptides.
   - LC-MS/MS Analysis: Analyze the fractions by liquid chromatography-tandem mass spectrometry (LC-MS/MS) to identify and quantify AMPD peptides.
   - Internal Standard Addition: Spike the digested samples with the isotopically labeled AMPD peptides at known concentrations.
   - Data Acquisition: Collect MS/MS data to identify and quantify both endogenous AMPD peptides and their labeled counterparts.
   - Quantification: Use the peak area ratios of heavy to light peptides to calculate the absolute quantity of AMPD peptides in the samples.
   - Bioinformatics Analysis: Employ software to process raw data, identify peptides, and perform quantitative analysis.
   - Statistical Validation: Perform statistical analysis to validate the upregulation of AMPD in the high fructose diet group compared to the control.

### Question 5
<details>
  <summary>**Q5**:</summary>
  ![](images/BIO312 final exam q5 (2021-2022).jpg)
</details>
Solution:

- Genome Analysis: Analyze the genome sequence of bacterium A to identify genes encoding for the secreted toxins responsible for B disease.
- Protein Identification: Use bioinformatics tools to predict the open reading frames (ORFs) and identify potential toxin proteins.
- Gene Cloning: Clone the identified toxin genes into an expression vector suitable for protein production in a host organism (e.g., E. coli or yeast).
- Protein Expression: Express the recombinant toxin proteins in the host organism, ensuring proper folding and post-translational modifications if necessary.
- Protein Purification: Purify the expressed proteins using techniques such as affinity chromatography, ion exchange chromatography, or gel filtration.
- Immunogenicity Assessment: Test the purified proteins for their ability to induce an immune response using in vitro assays or animal models.
- Adjuvant Selection: Select an appropriate adjuvant to enhance the immune response to the vaccine antigens.
- Formulation Development: Formulate the vaccine by combining the purified proteins with the selected adjuvant to create a stable and effective vaccine formulation.
- Preclinical Testing: Conduct extensive preclinical testing in animal models to evaluate the safety, efficacy, and immunogenicity of the vaccine.
- Clinical Trials: Proceed with clinical trials in humans, starting with Phase I to assess safety, followed by Phase II to evaluate immunogenicity and optimal dosing, and Phase III for efficacy.

### Question 6
<details>
  <summary>**Q6**:</summary>
  ![](images/BIO312 final exam q6 (2021-2022).jpg)
</details>
Solution:

1. Protein X is rich in alpha helix, and protein Y appears as a random coil.
2. Reasons:
   - 190 nm Region: The higher ellipticity in the combined spectrum (X+Y) compared to when both proteins are together (X:Y) at around 190 nm could indicate that the individual proteins have a higher beta-sheet content or are more aggregated when separate. This peak is often associated with the presence of beta-sheets or aggregated structures.
   - 220 nm Region: The higher ellipticity in the X:Y spectrum compared to the combined individual spectra (X+Y) at around 220 nm suggests that when proteins X and Y are mixed together, there is either an increase in alpha-helical content or a stabilization of the alpha-helical structure. This could be due to a structural change upon interaction, or the formation of a complex that stabilizes the helices.
3. The alpha-helical content of protein X may be reduced due to interaction with protein Y. Non-specific interactions such as electrostatic or hydrophobic forces might cause structural loosening in protein X.
4. Several techniques can be used to study protein-protein or protein-ligand binding and determine their stoichiometry:
   - Surface Plasmon Resonance (SPR): Measures changes in refractive index upon binding, providing real-time kinetics and binding stoichiometry.
   - Isothermal Titration Calorimetry (ITC): Measures heat changes during binding, allowing for the determination of binding affinity and stoichiometry.
   - Fluorescence Resonance Energy Transfer (FRET): Uses energy transfer between a donor and acceptor to study binding and proximity, which can indirectly provide stoichiometry.
   - Analytical Ultracentrifugation: Monitors sedimentation behavior to infer binding and complex formation.

### Question 7
<details>
  <summary>**Q7**:</summary>
  ![](images/BIO312 final exam q7 (2021-2022).jpg)
</details>
Solution:

1. The domain type of triose phosphate isomerase is the TIM barrel (also known as Œ±/Œ≤-barrel). This structural motif consists of eight parallel beta-strands forming a barrel-like structure with alpha-helices running along the outside.
2. Technique to Characterize Secondary Structure:
   - Circular Dichroism (CD) Spectroscopy: To determine the protein's secondary structure content based on the differences in the way it interacts with circularly polarized light.
   - Infrared (IR) Spectroscopy: Particularly Attenuated Total Reflectance (ATR)-FTIR, which can provide information about the secondary structure by measuring the amide bonds' vibrations.
   - Nuclear Magnetic Resonance (NMR) Spectroscopy: If the protein is small enough or suitable isotopes are used, NMR can provide detailed structural information.
3. Molecular replacement is used to solve the structure of triose phosphate isomerase because it is a method that allows the prediction of the three-dimensional structure of a protein when a similar protein's structure is already known. It involves placing a known structure (the search model) into the unknown density map and then refining its position.<br>
An alternative method to solve a protein crystal structure is Ab initio phasing. When no similar structure is available, this method involves using the diffraction data to build an electron density map from scratch, often starting with small molecular fragments and building up the entire structure.

### Question 8a
<details>
  <summary>**Q8a**:</summary>
  ![](images/BIO312 final exam q8a (2021-2022).jpg)
</details>
Solution:

Crystal mosaic spread, refers to the variation in orientation of small crystallites within a larger crystal. This results in a diffraction pattern where individual spots are broadened and their intensities are reduced due to the averaging effect of the slightly misaligned crystallites. Consequently, mosaicity can cause inaccurate X-ray intensity measurements because the overlapping and weakened spots make it challenging to precisely determine the position and intensity of each reflection, leading to errors in data analysis.

### Question 8b
<details>
  <summary>**Q8b**:</summary>
  ![](images/BIO312 final exam q8b (2021-2022).jpg)
</details>
Solution:

NOSEY (Nuclear Overhauser Effect Spectroscopy) is an NMR technique that measures the Overhauser effect between nuclear spins to determine proximity relationships between atoms in a molecule. It provides distance constraints used to infer the 3D structure of proteins.<br>
_(During a NOSEY experiment, the magnetization of a nucleus is transferred to a neighboring nucleus via the Overhauser effect. By measuring the intensity of these transferred signals, researchers can map out the network of spatially close atoms within the protein. This information is then used in conjunction with other NMR data to calculate the three-dimensional structure of the protein through computational methods such as molecular dynamics simulations or distance geometry calculations.)_

# [SessionInfo]{style="color:green"}
```{r, echo=FALSE}
sessionInfo()
```